languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -43,11 +46,31 @@ def clear
       # @param name [String]
       # @raise [PolarFileExtensionError] if provided filename has invalid extension.
       # @raise [PolarFileNotFoundError] if provided filename does not exist.
-      def load_file(name)
+      def load_file(name) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         raise PolarFileExtensionError unless ['.pol', '.polar'].include? File.extname(name)
-        raise PolarFileNotFoundError, name unless File.file?(name)
 
-        load_queue << name
+        begin
+          file = File.open(name)
+          file_data = file.read
+          hash = Digest::MD5.hexdigest(file_data)
+          file.close
+        rescue Errno::ENOENT
+          raise PolarFileNotFoundError, name unless File.file?(name)
+        end
```

Comment:
nit: I would structure this as
```suggestion
        content = File.open(name) { |f| f.read }
        hash = Digest::MD5.hexdigest(content)
```
and then down at the end of the method you can `rescue` like this:
```rb
def my_method
  ...
rescue Errno::ENOENT
  raise PolarFileNotFoundError, name
end
```